proposals/p3833.md

```diff
@@ -112,6 +113,34 @@ understand how the design works in complex cases and to debug unexpected
 behavior in the toolchain. This means starting with that implementation doesn't
 result in sunk cost.
 
+### Example: overloaded operators
+
+As an example to help illustrate this, how should the toolchain model overloaded
+operators? This proposal suggests that _initially_, the implementation should
+aim to fully model the rewrite-based dispatch through interfaces in the prelude.
+That is, each use af `x + y` should turn into roughly equivalent SemIR as would
+be used to model the rewritten semantics of `x.(Core.AddWith(typeof(y)).Op)(y)`.
+This in turn would dispatch to an exact-type implementation which would provide
+any implicit conversions, and so-on.
```

Comment:
Could you explain the alternative(s) that is/are being decided against in terms of this example as well?

Reply:
Maybe we could point out that this happens even when adding together (say) `i32`s.

Reply:
Ah, that makes sense, I think done.